mrmustard/physics/ansatz/polyexp_ansatz.py

```diff
@@ -934,10 +929,12 @@ def __call__(self: PolyExpAnsatz, *z_inputs: ArrayLike | None) -> Batch[ComplexT
 
         exp_sum = self._compute_exp_part(z, A, b)
         if self.num_derived_vars == 0:  # purely gaussian
-            return math.einsum("...,...->...", exp_sum, c)
+            ret = math.einsum("...,...->...", exp_sum, c)
         else:
             poly = self._compute_polynomial_part(z, A, b)
-            return self._combine_exp_and_poly(exp_sum, poly, c)
+            ret = self._combine_exp_and_poly(exp_sum, poly, c)
+
+        return math.sum(ret, axis=-1) if self._lin_sup else ret
```

Comment:
Still feels a little odd that this kind of thing is checked in the `return` instead of e.g. within `combin_exp_and_poly`, but I don't yet have enough experience with the code base to understand reasons why or why not to keep it there.